### PR TITLE
[AMS-1166] Cloning tool: added support for color and logo elements

### PR DESCRIPTION
**Pull request by tukaef**

### CODE REVIEW

File: src/CloningTool/CloneStrategies/CloneAdvertisementsBase.cs [hunk 1]
```diff
@@ -281,13 +281,11 @@ public abstract class CloneAdvertisementsBase : ICloneStrategy
                                  .Select(async cutomImage =>
```
Comment (xakep139) [inline src/CloningTool/CloneStrategies/CloneAdvertisementsBase.cs]: Typo

File: src/CloningTool/RestClient/OkapiClient.cs [hunk 2]
```diff
@@ -725,11 +725,18 @@ public void Dispose()
                         using (var streamContent = new StreamContent(memoryStream))
                         {
                             content.Add(streamContent, fileName, fileName);
-                            using (var response = await _authorizedHttpClient.PostAsync(url, content))
+                            var request = new HttpRequestMessage(HttpMethod.Post, url) { Content = content };
```
Comment (xakep139) [inline src/CloningTool/RestClient/OkapiClient.cs]: `HttpRequestMessage` is disposable

File: src/CloningTool/CloneStrategies/CloneAdvertisementsBase.cs [hunk 3]
```diff
@@ -250,26 +251,60 @@ public abstract class CloneAdvertisementsBase : ICloneStrategy
...
-                                                                   fileData);
-                    Interlocked.Increment(ref _uploadedBinariesCount);
-                    element.Value = res;
+                    continue;
+                }
+
+                var fileData = await SourceRestClient.DownloadFileAsync(sourceDescriptor.Id, value.DownloadUri);
+                var matchedElement = newAdvertisement.Elements.First(e => e.TemplateCode == element.TemplateCode);
+                EnsureUploadUrlIsValid(sourceDescriptor.Id, matchedElement);
+                var uploadResponse = await DestRestClient.UploadFileAsync(sourceDescriptor.Id,
```
Comment (denisivan0v) [inline src/CloningTool/CloneStrategies/CloneAdvertisementsBase.cs]: It's better to move all the params on next lines, including the first one.

File: src/CloningTool/RestClient/OkapiClient.cs [hunk 4]
```diff
@@ -707,7 +707,7 @@ public void Dispose()
             }
         }
 
-        public async Task<IObjectElementValue> UploadFileAsync(long advertisementId, Uri uploadUrl, string fileName, byte[] fileData)
+        public async Task<ApiObjectElementRawValue> UploadFileAsync(long advertisementId, Uri uploadUrl, string fileName, byte[] fileData, IFileUploadParams fileUploadParams)
         {
```
Comment (denisivan0v) [inline src/CloningTool/RestClient/OkapiClient.cs]: See comment on `IFileUploadParams` and HTTP headers in #126

File: src/CloningTool/CloneStrategies/CloneAdvertisementsBase.cs (hunk 3 above)
Comment (tukaef) [reply src/CloningTool/CloneStrategies/CloneAdvertisementsBase.cs]: Done

File: src/CloningTool/RestClient/OkapiClient.cs (hunk 4 above)
Comment (tukaef) [reply src/CloningTool/RestClient/OkapiClient.cs]: Removed `IFileUploadParams` from here

File: src/CloningTool/RestClient/OkapiClient.cs (hunk 2 above)
Comment (tukaef) [reply src/CloningTool/RestClient/OkapiClient.cs]: Fixed

**Pull request merged**